croptalk/tools.py

```diff
@@ -1,52 +1,111 @@
+    Words: {question}
+    commodity: """
 
-    Returns: the tool returns a string which gives the coverage level with their
-            associated percentage indemnified statistic
+    commodity_prompt = PromptTemplate.from_template(commodity_template_tool)
+    return commodity_prompt | llm | StrOutputParser()
 
+
+@tool("get_SP_doc")
+def get_sp_document(state: Optional[str] = None,
+                    county: Optional[str] = None,
+                    commodity: Optional[str] = None,
+                    year: Optional[int] = None):
     """
+    This tool is used to query the SP document from a database along state, county, commodity and year.
+    For instance, a user might ask :
+    - find me the SP document related to Oranges in Yakima, Washington for the year 2024
+    - SP document for corn in Butte, California, 2022
 
-    sob = SOB[SOB["commodity_year"] == 2023]
+    Args:
+    state : name of state
+    county_name: name of county provided
+    commodity_name: name of commodity
+    insurance_plan_name: provided plan name abbreviation
```

Comment:
Looks like `insurance_plan_name` should be replaced by `year` in docstring based on function's signature, no? ;)

Reply:
I think you might be right!